src/fsharp/FSharp.Build/FSharpEmbedResXSource.fs

```diff
@@ -0,0 +1,144 @@
+
+namespace Microsoft.FSharp.Build
+
+open System
+open System.Collections
+open System.Globalization
+open System.IO
+open System.Linq
+open System.Text
+open System.Xml.Linq
+open Microsoft.Build.Framework
+open Microsoft.Build.Utilities
+
+type FSharpEmbedResXSource() =
+    let mutable _buildEngine : IBuildEngine = null
+    let mutable _hostObject : ITaskHost = null
+    let mutable _embeddedText : ITaskItem[] = [||]
+    let mutable _generatedSource : ITaskItem[] = [||]
+    let mutable _outputPath : string = ""
+    let mutable _targetFramework : string = ""
+
+    let boilerplate = @"// <auto-generated>
+
+namespace {0}
+
+open System.Reflection
+
+module internal {1} =
+    type private C (_dummy:System.Object) = class end
+    let mutable Culture = System.Globalization.CultureInfo.CurrentUICulture
+    let ResourceManager = new System.Resources.ResourceManager(""{2}"", C(null).GetType().GetTypeInfo().Assembly)
+    let GetString(name:System.String) : System.String = ResourceManager.GetString(name, Culture)"
+
+    let boilerplateGetObject = "    let GetObject(name:System.String) : System.Object = ResourceManager.GetObject(name, Culture)"
+
+    let generateSource (resx:string) (fullModuleName:string) (generateLegacy:bool) (generateLiteral:bool) =
+        try
+            let printMessage = printfn "FSharpEmbedResXSource: %s"
+            let justFileName = Path.GetFileNameWithoutExtension(resx)
+            let namespaceName, moduleName =
+                let parts = fullModuleName.Split('.')
+                if parts.Length = 1 then ("global", parts.[0])
+                else (String.Join(".", parts, 0, parts.Length - 1), parts.[parts.Length - 1])
+            let generateGetObject =
+                match _targetFramework with
+                | "netstandard1.0"
+                | "netstandard1.1"
+                | "netstandard1.2"
+                | "netstandard1.3"
+                | "netstandard1.4"
+                | "netstandard1.5"
+                | "netstandard1.6" -> false // these targets don't support the `ResourceManager.GetObject()` method
+                | _ -> true // other supported runtimes, do
+            let sb = StringBuilder().AppendLine(String.Format(boilerplate, namespaceName, moduleName, justFileName))
+            if generateGetObject then sb.AppendLine(boilerplateGetObject) |> ignore
+            let sourcePath = Path.Combine(_outputPath, justFileName + ".fs")
+            printMessage <| sprintf "Generating: %s" sourcePath
+            let body =
+                let xname = XName.op_Implicit
+                XDocument.Load(resx).Descendants(xname "data")
+                |> Seq.fold (fun (sb:StringBuilder) (node:XElement) ->
+                    let name =
+                        match node.Attribute(xname "name") with
+                        | null -> failwith "Missing resource name"
```

Comment:
Can we provide a more helpful message here?

Reply:
Fixed.  Updated commits coming soon.